pkg/event/target/kafka.go

```diff
@@ -237,6 +237,8 @@ func (k KafkaArgs) pingBrokers() bool {
 func NewKafkaTarget(id string, args KafkaArgs, doneCh <-chan struct{}, loggerOnce func(ctx context.Context, err error, id interface{}, kind ...interface{})) (*KafkaTarget, error) {
 	config := sarama.NewConfig()
 
+	config.Version = sarama.V0_10_0_0
```

Comment:
@kannappanr i read this in their design documentation. Setting it to a higher value may break the older ones.

```
// The version of Kafka that Sarama will assume it is running against.
// Defaults to the oldest supported stable version. Since Kafka provides
// backwards-compatibility, setting it to a version older than you have
// will not break anything, although it may prevent you from using the
// latest features. Setting it to a version greater than you are actually
// running may lead to random breakage.
```

IMO, we could accept the version as a config key, and not setting `config.Version` by default

Reply:
We could use https://github.com/Shopify/sarama/blob/master/utils.go#L196 to validate and parse the version string provided by the user while setting